account_easy_reconcile/easy_reconcile.py

```diff
@@ -0,0 +1,424 @@
+
+    @api.one
+    @api.depends('account')
+    def _get_total_unrec(self):
+        obj_move_line = self.env['account.move.line']
+        self.unreconciled_count = len(obj_move_line.search(
+            [('account_id', '=', self.account.id),
+             ('reconcile_id', '=', False),
+             ('reconcile_partial_id', '=', False)],
+            ))
+
+    @api.one
+    @api.depends('account')
+    def _get_partial_rec(self):
+        obj_move_line = self.env['account.move.line']
+        self.reconciled_partial_count = len(obj_move_line.search(
+            [('account_id', '=', self.account.id),
+             ('reconcile_id', '=', False),
+             ('reconcile_partial_id', '!=', False)],
+            ))
+
+    @api.one
+    @api.depends('history_ids')
+    def _last_history(self):
+        # do a search() for retrieving the latest history line,
+        # as a read() will badly split the list of ids with 'date desc'
+        # and return the wrong result.
+        history_obj = self.env['easy.reconcile.history']
+        last_history_rs = history_obj.search(
+            [('easy_reconcile_id', '=', self.id)],
+            limit=1, order='date desc'
+            )
+        self.last_history = last_history_rs[0] if last_history_rs else False
+
+    name = fields.Char(string='Name', size=32, required=True)
+    account = fields.Many2one('account.account',
+                              string='Account',
+                              required=True,
+                              )
+    reconcile_method = fields.One2many('account.easy.reconcile.method',
+                                       'task_id',
+                                       string='Method'
+                                       )
+    unreconciled_count = fields.Integer(string='Unreconciled Items',
+                                        compute='_get_total_unrec'
+                                        )
+    reconciled_partial_count = fields.Integer(
+        string='Partially Reconciled Items',
+        compute='_get_partial_rec'
+        )
+    history_ids = fields.One2many('easy.reconcile.history',
+                                  'easy_reconcile_id',
+                                  string='History',
+                                  readonly=True
+                                  )
+    last_history = fields.Many2one('easy.reconcile.history',
+                                   string='readonly=True',
+                                   compute='_last_history',
+                                   readonly=True
+                                   )
+    company_id = fields.Many2one('res.company', string='Company')
+
+    @api.model
+    def _prepare_run_transient(self, rec_method):
+        return {'account_id': rec_method.task_id.account.id,
+                'write_off': rec_method.write_off,
+                'account_lost_id': (rec_method.account_lost_id and
+                                    rec_method.account_lost_id.id),
```

Comment:
in new API a_model.m2o_field.id will return False is m2o_field is an empty record set so no need of the `and`